Tests/MapboxMapsTests/Foundation/Extensions/Core/DownloadStatusTests.swift

```diff
@@ -0,0 +1,89 @@
+import XCTest
+@testable import MapboxMaps
+
+final class DownloadStatusTests: XCTestCase {
```

Comment:
I'm thinking there are maybe 9 tests to do here…

1. Test the first convenience initializer where httpResult is success. Use `__httpResult`, not `httpExpected` to verify the behavior so that we can isolate the initializer
2. Like 1, but where httpResult is failure
3. Like 1, but for the second convenience initializer
4. Like 2, but for the second convenience initializer
5. Test httpResult in the case where `__httpResult` is `nil`. Don't use the convenience constructors in this test so that we can better isolate httpResult's implementation
6. Test httpResult for the case where __httpResult is a value
7. Test httpResult for the case where __httpResult is an error
8. Test totalBytes when it's nil
9. Test totalBytes when it's non-nil